timesketch/api/v1/resources.py

```diff
@@ -1407,45 +1452,58 @@ def post(self):
         if not isinstance(filename, six.text_type):
             filename = codecs.decode(filename, 'utf-8')
 
-        index_name = uuid.uuid4().hex
+        index_name = form.index_name.data or uuid.uuid4().hex
         if not isinstance(index_name, six.text_type):
             index_name = codecs.decode(index_name, 'utf-8')
 
         file_path = os.path.join(upload_folder, filename)
         file_storage.save(file_path)
 
-        # Create the search index in the Timesketch database
-        searchindex = SearchIndex.get_or_create(
+        # Check if search index already exists.
+        searchindex = SearchIndex.query.filter_by(
```

Comment:
hm, why not use get_or_create() here?

Reply:
I don't want to create one here if it doesn't exist, only querying whether it exists or not.

Reply:
ack